app/CLI/Actions.hs

```diff
@@ -93,8 +102,13 @@ doOnMatchingStage terms requiredStage action = do
               <> show requiredStage
           )
 
-handleCliAction :: CliAction -> Runner ()
-handleCliAction userAction = do
+handleCliAction ::
+  (DelegateInterface.FrontendRequest -> IO ()) ->
+  IORef DelegateState ->
```

Comment:
@uhbif19 is there a specific reason why we use an IORef here instead of a state transformer?

Reply:
Because there is thread polling `DelegateState`.